From 1a1c32772a2a84afa7df5731559eee9301dc8fa4 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Fri, 14 Feb 2020 17:48:48 +0100
Subject: [PATCH] Add bash, busybox, and rescue as part of base modules
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Fabiano Fidêncio <fidencio@redhat.com>
---
 dracut/dracut.conf.d/05-base.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut/dracut.conf.d/05-base.conf b/dracut/dracut.conf.d/05-base.conf
index 1dd41c4..18d07b0 100644
--- a/dracut/dracut.conf.d/05-base.conf
+++ b/dracut/dracut.conf.d/05-base.conf
@@ -14,4 +14,4 @@ hostonly_cmdline="no"
 # create reproducible images
 reproducible="yes"
 # dracut modules to include (NOTE: these are NOT kernel modules)
-dracutmodules="kernel-modules udev-rules syslog systemd"
+dracutmodules="kernel-modules udev-rules syslog systemd bash busybox rescue"
-- 
2.24.1

